RecoHGCal/TICL/plugins/SeedingRegionByTracks.cc

```diff
@@ -0,0 +1,74 @@
+// Author: Arabella Martelli, Felice Pantaleo, Marco Rovere - arabella.martelli@cern.ch, felice.pantaleo@cern.ch, marco.rovere@cern.ch
+// Date: 06/2019
+#include <algorithm>
+#include <set>
+#include <vector>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SeedingRegionByTracks.h"
```

Comment:
plugins should not have separate headers

Reply:
Also in this case I see the same logic applied to RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc and .h 
The plugin/producer is RecoHGCal/TICL/plugins/TICLSeedingRegionProducer.cc and for this there is no .h

and same for RecoHGCal/TICL/plugins/SeedingRegionGlobal.cc

Reply:
you are not using a factory pattern. So, the header file is needed after all.